src/CodeGen_GPU_Dev.cpp

```diff
@@ -149,7 +149,12 @@ void CodeGen_GPU_C::visit(const Shuffle *op) {
             internal_assert(op->vectors[0].type() == op->vectors[i].type());
         }
         internal_assert(op->type.lanes() == (int)op->indices.size());
-        const int max_index = (int)(op->vectors[0].type().lanes() * op->vectors.size());
+        std::vector<int> vector_first_index;
```

Comment:
Comments needed for the logic behind how these indices are being generated, and what was wrong before